src/sympc/protocol/spdz/spdz.py

```diff
@@ -55,24 +55,27 @@ def mul_master(
     shape_x = tuple(x.shape)
     shape_y = tuple(y.shape)
 
-    CryptoPrimitiveProvider.generate_primitives(
-        f"beaver_{op_str}",
-        sessions=session.session_ptrs,
-        g_kwargs={
-            "a_shape": shape_x,
-            "b_shape": shape_y,
-            "nr_parties": session.nr_parties,
-            **kwargs_,
-        },
-        p_kwargs={"a_shape": shape_x, "b_shape": shape_y},
-    )
-
     args = [
         list(el) + [op_str]
         for el in zip(session.session_ptrs, x.share_ptrs, y.share_ptrs)
     ]
 
-    mask = parallel_execution(spdz_mask, session.parties)(args)
+    try:
+        mask = parallel_execution(spdz_mask, session.parties)(args)
+    except (KeyError, ValueError):
```

Comment:
raise a specific Exception to avoid catching other unexpected errors 
And don't forget to add tests :)

Reply:
done